src/Scorecard.js

```diff
@@ -0,0 +1,46 @@
+class Scorecard {
+  constructor() {
+    this.frames = []
+  }
+
+  getFrame(n) {
+    return this.frames[n]
+  }
+
+  getLastFrame() {
+    return this.frames[this.frames.length - 1]
+  }
+
+  getNumFrames() {
+    return this.frames.length
+  }
+
+  roll(n) {
+    this._checkGameOver()
```

Comment:
You could change this into a return statement to make sure you're not throwing errors for regular behaviour, I'm not sure if there's a way to extract that to another method though. When I tried to do it, it returned 'game over' to the roll method but didn't cause the roll method itself to return. Defo something to look into :)

Reply:
I really liked your approach on this, you're right the game being over is not an error